app/services/achiever/course/template.rb

```diff
@@ -76,6 +77,20 @@ def self.find_by_activity_code(activity_code)
     template
   end
 
+  # @return [Achiever::Course::Template|nil] nil if not found
+  def self.maybe_find_by_activity_code(activity_code)
+    templates = all
+    template = templates.find { |val| val.activity_code == activity_code.upcase }
+
+    return template if template
+
+    message = "Could not find template #{activity_code} in Smart Connector"
+    Rails.logger.error(message)
+    Sentry.capture_message(message, level: :error)
+
+    template
```

Comment:
```suggestion
    nil
```

I like the idea of explicitly returning something, but everything in my head is saying it should explicitly be nil as you're returning it early if it has a value.

Or, if you were to restructure the function like:
```rb
# @return [Achiever::Course::Template|nil] nil if not found
  def self.maybe_find_by_activity_code(activity_code)
    templates = all
    template = templates.find { |val| val.activity_code == activity_code.upcase }

    unless template
      message = "Could not find template #{activity_code} in Smart Connector"
      Rails.logger.error(message)
      Sentry.capture_message(message, level: :error)
    end

    template
  end
```

it would be preferable